src/menus/PluginMenus.cpp

```diff
@@ -28,10 +28,381 @@
+
+struct EffectsMenuGroupsHandler : XMLTagHandler
+{
+   struct EffectsHandler : public XMLTagHandler
+   {
+      std::optional<std::string> textContent;
+      std::vector<TranslatableString>& effects;
+
+      EffectsHandler(std::vector<TranslatableString>& effects) : effects(effects) { }
+
+      bool HandleXMLTag(const std::string_view& tag, const AttributesList& attrs) override { return true; }
+      void HandleXMLContent(const std::string_view& text) override { textContent = text; }
+      void HandleXMLEndTag(const std::string_view& tag) override
+      {
+         if(textContent.has_value() && tag == "Effect")
+         {
+            effects.emplace_back(TranslatableString { *textContent, {} });
+            textContent = {};
+         }
+      }
+      XMLTagHandler* HandleXMLChild(const std::string_view& tag) override
+      {
+         if(tag == "Effect")
+            return this;
+         return nullptr;
+      }
+
+   };
+
+   struct GroupHandler : public XMLTagHandler
+   {
+      std::optional<std::string> textContent;
+      std::unique_ptr<EffectsHandler> effectsHandler;
+      std::pair<TranslatableString, std::vector<TranslatableString>>& group;
+
+      GroupHandler(std::pair<TranslatableString, std::vector<TranslatableString>>& group) : group(group) { }
+
+      bool HandleXMLTag(const std::string_view& tag, const AttributesList& attrs) override { return true; }
+      void HandleXMLContent(const std::string_view& text) override { textContent = text; }
+      void HandleXMLEndTag(const std::string_view& tag) override
+      {
+         if(tag == "Name")
+         {
+            group.first = TranslatableString { *textContent, { } };
+            textContent = { };
+         }
+      }
+      XMLTagHandler* HandleXMLChild(const std::string_view& tag) override
+      {
+         if(tag == "Effects")
+         {
+            effectsHandler = std::make_unique<EffectsHandler>(group.second);
+            return &*effectsHandler;
+         }
+         if(tag == "Name")
+            return this;
+
+         return nullptr;
+      }
+   };
+
+   EffectsMenuGroups& groups;
+   std::unique_ptr<GroupHandler> groupHandler;
+
+   EffectsMenuGroupsHandler(EffectsMenuGroups& groups) : groups(groups) { }
+
+   bool HandleXMLTag(const std::string_view& tag, const AttributesList& attrs) override { return true; }
+
+   XMLTagHandler* HandleXMLChild(const std::string_view& tag) override
+   {
+      if(tag == "Group")
+      {
+         groups.resize(groups.size() + 1);
+         groupHandler = std::make_unique<GroupHandler>(groups.back());
+         return &*groupHandler;
+      }
+      return nullptr;
+   }
+};
+
+EffectsMenuGroups LoadEffectsMenuGroups(const wxString& path)
+{
+   EffectsMenuGroups result;
+   EffectsMenuGroupsHandler handler(result);
+
+   XMLFileReader reader;
+   reader.Parse(&handler, path);
+   return result;
+}
+
+enum class GroupBy
```

Comment:
The first commit, introducing this enum and much else, is a very pleasing refactoring.